### ISSUE TITLE
Gui to server disconnect button bug

### ISSUE THREAD

**Issue by jtalier** [masa-umich/Ground-Support-Software#102]
When actively connected to the server through the gui, pressing the disconnect button on the GUI properly disconnects on the gui side. However the server shows no disconnect in the log. Upon re-connecting you are instantly connected, then disconnected. (likely culprit is the gui terminating the connection before sending command to server. Or server not recognizing the gui did disconnect)

**Comment by jtalier** [comment]
Issue fixed on branch JackDoingSomeWork with commit 871da54f5d1c8da2e854b9abf424ca7c329fddb0

Will close once it is merged into master

**Comment by jtalier** [comment]
Merged in 2.0.0
